In [ ]:
import requests
import json

In [ ]:
url = "https://msh-appdata.cafef.vn/rest-api/api/v1/MatchPrice?symbol=VNM&date=20240913"
response = requests.get(url)

In [ ]:
# Phân tích dữ liệu JSON
parsed_data = response.json()

# Truy cập và in dữ liệu aggregates
print("Aggregates:")
for aggregate in parsed_data["aggregates"]:
    print(f"Price: {aggregate['price']}, Total Volume: {aggregate['totalVolume']}, Volume Percent: {aggregate['volPercent']}")

# Truy cập và in dữ liệu trade
print("\nTrade Data:")
for trade in parsed_data["data"]:
    print(f"Symbol: {trade['symbol']}, Trade Date: {trade['tradeDate']}, Basic Price: {trade['basicPrice']}, Price: {trade['price']}, Volume: {trade['volume']}, Total Volume: {trade['totalVolume']}, Total Value: {trade['totalValue']}")

In [ ]:
data = DataFrame(parsed_data["data"])
print(data.head)

In [ ]:
pl.scatter(data, data['price'], data['volume'])

In [1]:
from requests import get
from pandas import DataFrame

link = "https://s.cafef.vn/Ajax/PageNew/DataHistory/PriceHistory.ashx?Symbol=FPT&StartDate=01/01/2023&EndDate=01/01/2024"
parsed_data = get(link).json()['Data']
data = DataFrame(parsed_data['Data'])
print(data.head)

<bound method NDFrame.head of           Ngay  GiaDieuChinh  GiaDongCua        ThayDoi  KhoiLuongKhopLenh  \
0   29/12/2023         82.99        96.1  -0.5(-0.52 %)            1866600   
1   28/12/2023         83.42        96.6  -0.3(-0.31 %)            1196600   
2   27/12/2023         83.68        96.9  -0.3(-0.31 %)            1355900   
3   26/12/2023         83.94        97.2    1.2(1.25 %)            3359900   
4   25/12/2023         82.91        96.0    1.4(1.48 %)            2017000   
5   22/12/2023         81.70        94.6  -0.5(-0.53 %)            1275400   
6   21/12/2023         82.13        95.1    0.4(0.42 %)            1756800   
7   20/12/2023         81.78        94.7    0.1(0.11 %)            1349500   
8   19/12/2023         81.70        94.6    0.6(0.64 %)            1250400   
9   18/12/2023         81.18        94.0  -2.2(-2.29 %)            2266500   
10  15/12/2023         83.08        96.2      0(0.00 %)            1474300   
11  14/12/2023         83.08      

#### Tổng quan:
- Dữ liệu bao gồm thông tin về giá cổ phiếu từ ngày 04/12/2023 đến 08/10/2024.
- Có sự tăng giá đáng kể trong khoảng thời gian này.


#### Xu hướng giá:
- Giá mở cửa vào ngày 04/12/2023 là 80.83 (tương đương 93.6 sau điều chỉnh).
- Giá đóng cửa vào ngày 08/10/2024 là 133.80, cho thấy sự tăng trưởng mạnh mẽ.
- Tăng trưởng khoảng 43% trong khoảng thời gian này.

#### Biến động giá:
- Giá có xu hướng tăng dần từ tháng 12/2023 đến cuối tháng 12/2023.
- Có một số biến động trong tháng 12, với mức giảm đáng kể vào ngày 18/12/2023 (-2.29%).
- Giá cao nhất trong tháng 12/2023 là 97.2 vào ngày 26/12/2023.

#### Khối lượng giao dịch:
- Khối lượng giao dịch biến động từ khoảng 1.2 triệu đến 4 triệu cổ phiếu mỗi ngày.
- Ngày có khối lượng giao dịch cao nhất là 12/12/2023 với 4,084,900 cổ phiếu.


#### Điểm đáng chú ý:

- Có sự tăng giá đột biến từ 96.1 (29/12/2023) lên 133.80 (08/10/2024), tăng khoảng 39% trong khoảng 9 tháng.
- Khối lượng giao dịch vào ngày 08/10/2024 (2,108,600) cao hơn trung bình, cho thấy sự quan tâm của nhà đầu tư.


#### Kết luận:
- Cổ phiếu này có xu hướng tăng giá mạnh trong khoảng thời gian được phân tích.
- Có thể có những sự kiện hoặc thông tin quan trọng ảnh hưởng đến giá cổ phiếu, đặc biệt là trong khoảng thời gian từ cuối tháng 12/2023 đến tháng 10/2024.